### PR TITLE
Inline Function implementation [Go only]

### PR DESCRIPTION
**Pull request by wuilliam321**
> Written this way, Inline Function is simple. In general, it isn't. I could write pages on how to handle recursion, multiple return points, inlining a method into another object when you don't have accessors, and the like. The reason I don't is that if you encounter these complexities, you should't do this refactoring.
**Martin Fowler, Refactoring**

This inline function refactoring (115), only works with simple functions which body are clear enough to scream their intent.

|       | 0       | 1       | > 1|
| ----------- | ----------- | ----------- | ----------- |
| Caller arguments [(hover_me)](# "Arguments passed down to a function call")  | supported  | supported       | supported |
| Function parameters [(hover_me)](# "Parameters defined by a function declaration")  | supported  | supported       | supported |
| Returned values [(hover_me)](# "Some languages support return a multiple values") | supported | supported       | supported |
| Body statements [(hover_me)](# "Each function declaration body lines") | supported | supported       | supported |
| Receivers variables [(hover_me)](# "Assignments where you receive returned values") | supported | supported       | supported |
| Return statements [(hover_me)](# "The line(s) that returns something") | supported | supported       | X |

Language support:
 - [ ] go (partially)
 - [ ] others (TBI)

### CODE REVIEW

File: lua/refactoring/config/init.lua [hunk 1]
```diff
@@ -68,6 +68,7 @@ local default_extract_var_statements = {}
 ---@field print_var fun(opts: {statement:string, prefix:string , var:string}): string
 ---@field comment fun(statement: string): string
 ---@field constant fun(opts: constant_opts): string
+---@field var_declaration fun(opts: code_generation): string
```
Comment (wuilliam321) [inline lua/refactoring/config/init.lua:71]: Not sure about it, it changed in the last rebase I made

File: lua/refactoring/treesitter/langs/go.lua [hunk 2]
```diff
@@ -48,10 +48,25 @@ function Golang.new(bufnr, ft)
                 "(var_declaration (var_spec name: (identifier) @tmp_capture))"
             ),
         },
+        -- TODO: not in doc
```
Comment (wuilliam321) [inline lua/refactoring/treesitter/langs/go.lua:51]: Not sure how to add this to the doc!

**Comment by pranavrao145** [comment]
Hi @wuilliam321. Apologies for the delayed review and thank you for your contribution! Can you run `make pr-ready` on your branch, then commit and push? Tests look good, but this command will run styling and linting. Once this is done I'll proceed with a detailed review and get this merged ASAP. Thanks again for your contribution!

**Comment by TheLeoP** [comment]
I'm closing this since most of it's code was copied into #414. Thanks for the contribution, #wuilliam321 and I'm sorry for not being able to rebase this PR on top of master.

**Comment by wuilliam321** [comment]
Totally understand! I haven't had the time to rebase and/or implement more
languages.

**Pull request closed without merging**